### PR TITLE
[Op][NN2] BatchNorm, LayerNorm, Matmul, Dropout, CrossEntropy

### PR DESCRIPTION
**Pull request by MasterJH5574**
Following #67, this PR introduces the rest part of NN operators we already have to the latest StructInfo codebase. It is part of our operator migration efforts as listed in #62.

* Normalization: BatchNorm, LayerNorm
* Matmul
* Dropout
* Loss function: CrossEntropy, SoftmaxCrossEntropy

### CODE REVIEW

**Comment by MasterJH5574** [comment]
Followup PR #73

File: src/relax/op/op_common.cc [hunk 1]
```diff
@@ -45,6 +45,10 @@ Array<TensorStructInfo> GetInputTensorStructInfo(const Call& call, const BlockBu
   return input_tensor_sinfo;
 }
 
+template Array<TensorStructInfo> GetInputTensorStructInfo<std::function<const char*(int)>>(
+    const Call& call, const BlockBuilder& ctx, int n_input,
+    std::function<const char*(int)> f_get_input_names);
+
```
Comment (MasterJH5574) [inline src/relax/op/op_common.cc]: Added this in order to support lazy name construction, through the approach of lambdas. My only concern is that, I’m not sure if the lambdas will kept inlined in this case, where I instantiated the template function explicitly. I’m kinda doubting that.

File: src/relax/op/tensor/binary.cc [hunk 2]
```diff
@@ -58,7 +56,13 @@ namespace relax {
 template <typename FType>
 StructInfo InferStructInfoBroadcast(const Call& call, const BlockBuilder& ctx,
                                     FType f_compute_out_dtype) {
-  auto [lhs_sinfo, rhs_sinfo] = GetBinaryInputTensorStructInfo(call, ctx, /*op_name=*/"Binary");
+  Array<TensorStructInfo> input_sinfo = GetInputTensorStructInfo<std::function<const char*(int)>>(
+      call, ctx, /*n_input=*/2, [](int i) {
+        ICHECK(i == 0 || i == 1);
```
Comment (tqchen) [inline src/relax/op/tensor/binary.cc]: we can use https://github.com/tlc-pack/relax/blob/relax/include/tvm/ir/op.h#L70 which contains all the information needed

File: src/relax/op/op_common.cc (hunk 1 above)
Comment (tqchen) [reply src/relax/op/op_common.cc]: we can use https://github.com/tlc-pack/relax/blob/relax/include/tvm/ir/op.h#L70 which contains all the information needed

Comment (tqchen) [reply src/relax/op/op_common.cc]: if you want to eliminate lambda, you don't want to pass in template specializations, instead make the function an inline function and implcit pass in value by auto

**Comment by MasterJH5574** [comment]
@tqchen Thanks for the reminder on `OpNode::arguments`! Just updated with that. Now all places use `arguments` for the number of inputs and the input names.

**Review by tqchen — APPROVED** [review]
